store/mockstore/mocktikv/cluster.go

```diff
@@ -269,6 +270,49 @@ func (c *Cluster) GetRegionByID(regionID uint64) (*metapb.Region, *metapb.Peer)
 	return nil, nil
 }
 
+// ScanRegions returns at most `limit` regions from given `key` and their leaders.
+func (c *Cluster) ScanRegions(key []byte, limit int) ([]*metapb.Region, []*metapb.Peer) {
+	c.RLock()
+	defer c.RUnlock()
+
+	regions := make([]*Region, 0, len(c.regions))
```

Comment:
We only need `limit` regions but copy the entire slice here. It's OK because the mocktikv will not contain a lot of regions.

Reply:
The regions is got in arbitrary order and ScanRegions needs to scan regions by key. I get all regions here and do a sorting below. I think this will be faster than finding regions by key for at most `limit` times, because the mock cluster doesn't support quickly finding region by key. You can see that in `GetRegionByKey` method, it does a linear searching on `c.regions`.